File: leetcode/2024/26.py
''' 
given a sorted array size n = 10e4
let k be the number of unique elements in arr
modify the array such that the first k elements contain no duplicates
return k

helper data structures (e.g., sets) are not allowed
'''
class Solution:
    def solve(self, nums):
        k = 0
        for i in range(len(arr)):
            if (arr[i] != arr[k]):
                k += 1
                arr[k] = arr[i]
        return k + 1

test = Solution()

arr = [1, 1, 2, 2, 3, 3]
k = test.solve(arr)

print(arr)
print(k)

File: leetcode/2024/26.c
#include <stdio.h>

int solve(int* arr, int n, int val) {
	int k = 0;
	for (int i = 0; i < n; i++) {
		if (arr[i] != arr[k]) {
			k++;
			arr[k] = arr[i];
		}
	}

	return k + 1;
}


int main() {
	int arr[] = { 1, 1, 2, 2, 3, 3 };
	int n = 6;

	int k = solve(arr, n, 2);

	printf("%d\n", k);
	for (int i = 0; i < n; i++) {
		printf("%d ", arr[i]);
	}
	printf("\n");

}
